src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1060,16 +1061,20 @@ private static bool IsDynamicStatics(TypeDesc type)
             // Non-interface dispatches go through the vtable.
             else if (!targetMethod.OwningType.IsInterface)
             {
-                pResult->kind = CORINFO_CALL_KIND.CORINFO_VIRTUALCALL_VTABLE;
+                pResult->kind = CORINFO_CALL_KIND.CORINFO_VIRTUALCALL_STUB;
                 pResult->nullInstanceCheck = true;
 
                 // We'll special virtual calls to target methods in the corelib assembly when compiling in R2R mode, and generate fragile-NI-like callsites for improved performance. We
                 // can do that because today we'll always service the corelib assembly and the runtime in one bundle. Any caller in the corelib version bubble can benefit from this
                 // performance optimization.
-                if (!MethodInSystemVersionBubble(callerMethod) || !MethodInSystemVersionBubble(targetMethod))
+                /* TODO-PERF: uncommenting the conditional statement below enables VTABLE-based calls for Corelib
```

Comment:
Could you file an issue for this and note it in the comment? The perf improvement this gave CoreLib was necessary to let us use R2R instead of fragile NIs in Core 3.0.

Reply:
For sure, that's a good idea.

Reply:
Filed #7168 and verified it shows up as a "todo" in the project page. I'll add the bug ID to the code comment in the next iteration.

Reply:
The code comment has been updated to refer to the GitHub issue.